internal/generator/vector/output/metrics/prometheus.go

```diff
@@ -0,0 +1,77 @@
+package metrics
+
+import (
+	"github.com/openshift/cluster-logging-operator/internal/generator/framework"
+	"github.com/openshift/cluster-logging-operator/internal/generator/vector/helpers"
+)
+
+const (
+	AddNodenameToMetricTransformName = "add_nodename_to_metric"
+	PrometheusOutputSinkName         = "prometheus_output"
+	PrometheusExporterListenPort     = `24231`
+)
+
+type PrometheusExporter struct {
+	ID            string
+	Inputs        string
+	Address       string
+	TlsMinVersion string
+	CipherSuites  string
+}
+
+func (p PrometheusExporter) Name() string {
+	return "PrometheusExporterTemplate"
+}
+
+func (p PrometheusExporter) Template() string {
+	return `{{define "` + p.Name() + `" -}}
+[sinks.{{.ID}}]
+type = "prometheus_exporter"
+inputs = {{.Inputs}}
+address = "{{.Address}}"
+default_namespace = "collector"
+[sinks.{{.ID}}.tls]
+enabled = true
+key_file = "/etc/collector/metrics/tls.key"
+crt_file = "/etc/collector/metrics/tls.crt"
+min_tls_version = "{{.TlsMinVersion}}"
+ciphersuites = "{{.CipherSuites}}"
+{{end}}`
+}
+
+type AddNodenameToMetric struct {
```

Comment:
AddNodeNameToMetric is only used in one place, together with the PrometheusOutput, may be we could just inline it into `func (p PrometheusExporter) Template() string {` and remove all the  AddNodenameToMetric code?

Reply:
And even merge `internal_metrics` into that same template ^^^?

Reply:
sounds like a future refactoring